Tests/DatadogTests/Datadog/Kronos/KronosInternetAddressTests.swift

```diff
@@ -0,0 +1,111 @@
+            return [
+                try .mockIPv4([.mockRandom(otherThan: [10, 172, 192]), .mockRandom(), .mockRandom(), .mockRandom()]),
+                try .mockIPv4([172, .mockRandom(min: 0, max: 15), .mockRandom(), .mockRandom()]),
+                try .mockIPv4([172, .mockRandom(min: 32, max: 255), .mockRandom(), .mockRandom()]),
+                try .mockIPv4([192, .mockRandom(otherThan: [168]), .mockRandom(), .mockRandom()]),
+            ]
+        }
+
+        privateIPs.forEach { ip in
+            XCTAssertTrue(ip.isPrivate, "\(ip.host ?? "nil") should be private IP")
+        }
+        publicIPs.forEach { ip in
+            XCTAssertFalse(ip.isPrivate, "\(ip.host ?? "nil") should not be private IP")
+        }
+    }
+
+    func testIfIPv6AddressIsPrivarte() throws {
+        let privateIPs: [KronosInternetAddress] = try (0..<50).flatMap { _ in
+            return [
+                // random private IP starting with `fd` prefix
+                try .mockIPv6([0xfd] + (0..<15).map({ _ in .mockRandom() }))
+            ]
+        }
+        let publicIPs: [KronosInternetAddress] = try (0..<50).flatMap { _ in
+            return [
+                // first byte is mocked to avoid having `fd` prefix
+                try .mockIPv6([.mockRandom(min: 0xf0, otherThan: [0xfd])] + (0..<15).map({ _ in .mockRandom() })),
+                try .mockIPv6([.mockRandom(max: 0xfc, otherThan: [0xf])] + (0..<15).map({ _ in .mockRandom() })),
+            ]
+        }
+
+        privateIPs.forEach { ip in
+            XCTAssertTrue(ip.isPrivate, "\(ip.host ?? "nil") should be private IP")
+        }
+        publicIPs.forEach { ip in
+            XCTAssertFalse(ip.isPrivate, "\(ip.host ?? "nil") should not be private IP") // false-positive on `fd9:18d2:ef6e:e095:ffb6:69d8:dd8:22b9`
+        }
+    }
+}
+
+// MARK: - Mocks
+
+private extension KronosInternetAddress {
+    static func mockIPv4(_ bytes: [UInt8]) throws -> KronosInternetAddress {
+        precondition(bytes.count == 4, "Expected 4 bytes")
+        let numbers = bytes.map { String($0) }
+        let ipv4String = numbers.joined(separator: ".") // e.g. '192.168.1.1'
+        let address: KronosInternetAddress? = .mockWith(ipv4String: ipv4String)
+        return try XCTUnwrap(address, "\(ipv4String) is not a valid IPv4 string")
+    }
+
+    static func mockIPv6(_ bytes: [UInt8]) throws -> KronosInternetAddress {
+        precondition(bytes.count == 16, "Expected 16 bytes")
+        let groups: [String] = (0..<8).map { idx in
+            let hexA = String(bytes[idx * 2], radix: 16, uppercase: false)
+            let hexB = String(bytes[idx * 2 + 1], radix: 16, uppercase: false)
+            return hexA + hexB
+        }
+        let ipv6String = groups.joined(separator: ":") // e.g. 'ab:ab:ab:ab:ab:ab:ab:ab'
+        let address: KronosInternetAddress? = .mockWith(ipv6String: ipv6String)
+        return try XCTUnwrap(address, "\(ipv6String) is not a valid IPv6 string")
+    }
```

Comment:
The way we mock `KronosInternetAddress` values for IPv4 and IPv6 is quite tricky, but couldn't find a way to make it simpler. Alternatively, we could implement filtering as stand-alone function (`isPrivate(host:)`), but I found computed property (`ip.isPrivate`) more idiomatic.

Reply:
I'm not sure of the fuzzy tests value here 🤔 especially since the generated IPs is valid/expected we could simply test against edges and some pre-determined values in between, no?